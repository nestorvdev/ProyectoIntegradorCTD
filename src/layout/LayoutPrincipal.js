import Styles from "./styles.module.css";
import { useState } from "react";
import Header from "./Header";
import Footer from "./Footer";

export default function LayoutPrincipal(props) {
    const [showBurger, setShowBurger] = useState(false)

    return (
        <div className={showBurger === true ? Styles.opacity : null}>            
                <Header setLastLocation={props.setLastLocation} setBookingWithoutLogin={props.setBookingWithoutLogin} setLoading={props.setLoading} activeCreate={props.activeCreate} activeLogin={props.activeLogin} isLogged={props.isLogged} showBurger={showBurger} setShowBurger={setShowBurger} handleClean={props.handleClean} handleFavourite={props.handleFavourite}/>
                <main >
                    {props.children}
                </main>
                <Footer showBurger={showBurger} />           
        </div>
    )
}