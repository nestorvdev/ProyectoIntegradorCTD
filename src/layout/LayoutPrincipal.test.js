import React from 'react';
import {shallow, mount} from 'enzyme';
import '@testing-library/jest-dom';
import LayoutPrincipal from './LayoutPrincipal';

import Enzyme from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import { createSerializer } from 'enzyme-to-json';
Enzyme.configure({ adapter: new Adapter() });
expect.addSnapshotSerializer(createSerializer({ mode: 'deep' }));

describe("Probando el componente <LayoutPrincipal/>", () => {
    let wrapper;
    
    it("Deberia mostrar <LayoutPrincipal/> correctamente", () => {
        wrapper = shallow(<LayoutPrincipal/>)
        expect(wrapper).toMatchSnapshot();
    });

})