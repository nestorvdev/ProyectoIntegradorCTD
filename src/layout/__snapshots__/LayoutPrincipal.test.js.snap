// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Probando el componente <LayoutPrincipal/> Deberia mostrar <LayoutPrincipal/> correctamente 1`] = `
<div
  className={null}
>
  <Header
    setShowBurger={[Function]}
    showBurger={false}
  />
  <main />
  <Footer
    showBurger={false}
  />
</div>
`;
